src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -377,5 +384,30 @@ public virtual void EnableExtensions(bool enable = true)
             var rc = raw.sqlite3_enable_load_extension(_db, enable ? 1 : 0);
             SqliteException.ThrowExceptionForRC(rc, _db);
         }
+
+        /// <summary>
+        /// Raises the Microsoft.Data.Sqlite.SqliteConnection.Update event.
+        /// </summary>
+        /// <param name="e">A Microsoft.Data.Sqlite.UpdateEventArgs that contains the event data.</param>
+        protected virtual void OnUpdate(UpdateEventArgs e)
+            => Update?.Invoke(this, e);
+
+        private void UpdateHook(object user_data, int type, string database, string table, long rowid)
+            => OnUpdate(new UpdateEventArgs(ConvertToUpdateEventType(type), database, table, rowid));
+
+        private UpdateEventType ConvertToUpdateEventType(int type)
```

Comment:
Probably worth using the same underlying values in the enum so we can just cast (like `SqliteType`). I think this would also remove the need for `Unknown`.